target/bin/rspamd-dkim

```diff
@@ -0,0 +1,295 @@
+
+_trap_err_signal
+
+set -u -eE -o pipefail
+shopt -s inherit_errexit
+
+# shellcheck source=/dev/null
+source /etc/dms-settings
+
+function __usage
+{
+  _log 'trace' 'Showing usage message now'
+  echo -e "${PURPLE}RSPAMD-DKIM${RED}(${YELLOW}8${RED})
+
+${ORANGE}NAME${RESET}
+    rspamd-dkim - Configure DomainKeys Identified Mail (DKIM) via Rspamd
+
+${ORANGE}SYNOPSIS${RESET}
+    setup config dkim [ OPTIONS${RED}...${RESET} ]
+
+${ORANGE}DESCRIPTION${RESET}
+    This script aids in creating DKIM signing keys. The keys are created and managed by Rspamd.
+    OPTIONS can be used to configure a more complex setup.
+
+${ORANGE}OPTIONS${RESET}
+    ${BLUE}Generic Program Information${RESET}
+        -v        Enable verbose logging (setting the log level to 'debug').
+        -vv       Enable very verbose logging (setting the log level to 'trace').
+        help      Print the usage information.
+
+    ${BLUE}Configuration adjustments${RESET}
+        keytype   Set the type of key you want to use
+                  Possible values: rsa, ed25519
+                  Default: rsa
+        keysize   Set the size of the keys to be generated
+                  Possible values: 1024, 2048 and 4096
+                  Default: 2048
+                  Only applies when using keytype=rsa
+        selector  Set a manual selector for the key
+                  Default: mail
+        domain    Provide the domain for which keys are to be generated
+                  Default: primary domain name of DMS
+
+${ORANGE}EXAMPLES${RESET}
+    ${LWHITE}setup config dkim keysize 2048${RESET}
+        Creates keys of length 2048 bit in a default setup where domains are obtained from
+        your accounts.
+
+    ${LWHITE}setup config dkim keysize 512 selector 2023-dkim${RESET}
+        Creates keys of length 512 bit in a default setup where domains are obtained from
+        your accounts. The DKIM selector used is '2023-dkim'.
+
+    ${LWHITE}setup config dkim keysize 1024 selector 2023-dkim domain whoami.com${RESET}
+        Creates keys of length 1024 bit in a default setup where domains are obtained from your accounts.
+        The DKIM selector used is '2023-dkim'. The domain for which DKIM keys are generated is whoami.com.
+
+${ORANGE}EXIT STATUS${RESET}
+    Exit status is 0 if command was successful. If wrong arguments are provided or arguments contain
+    errors, the script will exit early with a non-zero exit status.
+
+"
+}
+
+function __do_as_rspamd_user
+{
+  local COMMAND=${1:?Command required when using __do_as_rspamd_user}
+  _log 'trace' "Running '${*}' as user '_rspamd' now"
+  shift 1
+  su -l '_rspamd' -s "$(command -v "${COMMAND}")" -- "${@}"
+}
+
+function _parse_arguments
+{
+  KEYTYPE='rsa'
+  KEYSIZE='2048'
+  SELECTOR='mail'
+  DOMAIN=${DOMAINNAME}
+
+  _log 'trace' "Options given to this script: '${*}'"
+
+  while [[ ${#} -gt 0 ]]
+  do
+    case "${1}" in
+
+      ( 'keytype' )
+        if [[ -n ${2+set} ]]
```

Comment:
Personally I find this ("assigning" a value) confusing on first view. ${2:-} work just fine (and is shorter).

Reply:
In this case, you're right, both work just fine. In cases where you'd require quotes though the `${...+sometext}` syntax is more robust. Due to the way that Bash's `[[ ]]` work, it'd be fine to use `:-` here instead of `+`, and I agree, it is easier to grasp.

I can rework that.